src/rules/encodingRule.ts

```diff
@@ -66,7 +66,7 @@ function showEncoding(encoding: Encoding): string {
 function detectEncoding(fileName: string): Encoding {
     const fd = fs.openSync(fileName, "r");
     const maxBytesRead = 3; // Only need 3 bytes to detect the encoding.
-    const buffer = new Buffer(maxBytesRead);
+    const buffer = Buffer.allocUnsafe(maxBytesRead);
```

Comment:
@NaridaL is there a not unsafe alternative? i'm not familiar with these internals.